packages/relayer/src/services/fee.ts

```diff
@@ -0,0 +1,279 @@
+      price_token_fee: rawTokenFee,
+      currencyContractId: params.instanceId,
+      exp: getExpirationTime(),
+      reciver_storage:
+        nearStoragePrice !== "0" ? params.receiverAccountId : null,
+    },
+    env.PRIVATE_KEY
+  );
+
+  return {
+    status: successStatus,
+    body: {
+      token,
+      status: "sucess",
+      timestamp: Date.now(),
+      valid_fee_for_ms: 120000,
+      percentage_fee: humanFee,
+      price_token_fee: rawTokenFee,
+    },
+  };
+};
+
+export const getExpirationTime = () => {
+  const nowInSeconds = Math.floor(Date.now() / 1000);
+
+  const twoMinutesInSeconds = 120;
+
+  return nowInSeconds + twoMinutesInSeconds;
+};
+
+export const validateFeeRequest = async (
+  { instanceId, receiverAccountId }: RequestParamsInterface,
+  { RPC_URL, HYC_CONTRACT }: Env
+): Promise<boolean> => {
+  const instanceIsAllowed = await viewFunction(
+    RPC_URL,
+    HYC_CONTRACT,
+    "view_is_contract_allowed",
+    {
+      account_id: instanceId,
+    }
+  );
+
+  const isValidReceiverAccountId = await checkIsValidAccountId(
+    receiverAccountId,
+    RPC_URL
+  );
+
+  return instanceIsAllowed && isValidReceiverAccountId;
+};
+
+/**
+ *
+ * @param accountId
+ * @returns
+ */
+export const checkIsValidAccountId = async (
+  accountId: string,
+  RPC_URL: string
+) => {
+  const regExpCheck =
+    /^(([a-z\d]+[\-_])*[a-z\d]+\.)*([a-z\d]+[\-_])*[a-z\d]+$/.test(accountId);
+
+  const checkLength = accountId.length > 2 && accountId.length <= 64;
+
+  const isRegisteredAccount = await checkIsRegisteredAccountId(
+    accountId,
+    RPC_URL
+  );
+
+  return regExpCheck && checkLength && isRegisteredAccount;
+};
+
+export const checkIsRegisteredAccountId = async (
+  accountId: string,
+  RPC_URL: string
+): Promise<boolean> => {
+  const isNammedAccount = accountId.includes(".");
+
+  if (!isNammedAccount) {
+    return true;
+  }
+
+  try {
+    await viewState(RPC_URL, accountId);
+
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+export const getCurrencyOfInstance = async (
+  instanceId: string,
+  { RPC_URL }: Env
+): Promise<{ depositValue: string; tokenId: string }> => {
+  const { currency, deposit_value } = await viewFunction(
+    RPC_URL,
+    instanceId,
+    "view_contract_params"
+  );
+
+  return {
+    depositValue: deposit_value,
+    tokenId: currency.account_id || "near",
+  };
+};
+
+export const getTokenDataById = async (token: string): Promise<any> => {
+  const res = await fetch(`https://api.coingecko.com/api/v3/coins/${token}`);
+
+  const { tickers } = (await res.json()) as any;
+
+  return tickers.find(({ target }: { target: string }) => target === "USD");
+};
+
+export const getNearStorageBoundsById = async (
+  receiverId: string,
+  currencyId: string,
+  env: Env
+): Promise<string> => {
+  const currentStorage = await getTokenStorage(
+    currencyId,
+    receiverId,
+    env.RPC_URL
+  );
+
+  if (!currentStorage || currentStorage.total < "0.10") {
```

Comment:
No need to check `currentStorage.total < "0.10"`. Only needs to check if account is registered